File: src/backend/reservation/src/main/java/org/taonaw/reservation/domain/model/member/MemberName.java
package org.taonaw.reservation.domain.model.member;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;

@Getter
@EqualsAndHashCode
@AllArgsConstructor
public class MemberName {
    @NonNull
    private final String firstName;
    @NonNull
    private final String lastName;

    public String asFormattedName() {
        return lastName + " " + firstName;
    }
}

File: src/backend/identityaccess/src/main/kotlin/org/taonaw/identityaccess/domain/model/memberAccount/MemberName.kt
package org.taonaw.identityaccess.domain.model.memberAccount

data class MemberName(
        val firstName: String,
        val lastName: String) {

    companion object {
        const val FIRST_NAME_LENGTH_MIN: Int = 1
        const val FIRST_NAME_LENGTH_MAX: Int = 20
        const val LAST_NAME_LENGTH_MIN: Int = 1
        const val LAST_NAME_LENGTH_MAX: Int = 20
    }

    init {
        require(lastName.length in LAST_NAME_LENGTH_MIN..LAST_NAME_LENGTH_MAX) {
            "姓 $LAST_NAME_LENGTH_MIN 以上 $LAST_NAME_LENGTH_MAX 以下でなければなりません。"
        }
        require(firstName.length in FIRST_NAME_LENGTH_MIN..FIRST_NAME_LENGTH_MAX) {
            "名は $FIRST_NAME_LENGTH_MIN 以上 $FIRST_NAME_LENGTH_MAX 以下でなければなりません。"
        }
    }

    fun asFormattedName(): String? {
        return "$lastName $firstName"
    }
}